java-frontend/src/main/java/org/sonar/java/model/expression/ConditionalExpressionTreeImpl.java

```diff
@@ -96,4 +98,10 @@ public void accept(TreeVisitor visitor) {
     );
   }
 
+  @Override
+  public void setInferedType(Type type) {
+    super.setInferedType(type);
+    if(((JavaType) trueExpression.symbolType()).isTagged(JavaType.DEFERRED)) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either remove or fill this block of code. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS00108)